Added man page, doc file

BobWall23 opened the pull request:
Added ironsftp.1 and README.md files.

BobWall23 committed:
Added man page, doc file

Added ironsftp.1 and README.md files.

BobWall23 committed:
Updated, license, copyright

Changed license in new source files from ISC to 3-clause BSD.
Updated copyright to add , Inc. after Labs

zmre committed:
Packaging (#37)

* Update build files for packaging

Updated configure.ac to have the AC_PROG_CC_STDC macro, which uses std99 and stops errors like, " ‘for’ loop initial declarations are only allowed in C99 mode"

Updated Makefile to install iron files

* Add RPM build capabilities

- Fixes #23
- Adds redhat build capabilities for centos7 fedora22 fedora23 fedora24 and fedora-rawhide

* Fixing the ruby proclivities of travis

* Add debian and ubuntu package support

Fixes #24

* Until upstream merges my PRs, use my fixed build repo

* code review fixes

zmre commented:
The travis fail is because these files do not include the packaging code and the .travis.yml file.

BobWall23 committed:
Print list of recipients when putting file (#44)

When uploading a file, print out a list of recipients with whom the file
was shared (excluding the current user).

Also fixed interactive prompt for ironsftp.

BobWall23 committed:
Fix signature calculation problem. (#43)

If an RSA signature is computed for a hash and at least eight of the
high order bits are zero, when the signature gets converted to a bignum,
those bits get dropped. When that happens, the length of the data in the
data signature packet ends up being shorter than the length we put in
the header. That was detected and reported as an "internal error".

To fix, removed that check, and instead added padding of zero bytes to
make the length of the data signature body match the size in the packet
header. It was necessary to do this rather than just write the extra
byte(s) of zeros at the start of the Multi-Precision Integer (MPI)
because then libgcrypt hated it. However, after doing the padding, the
signature woudln't verify, because the OpenSSL RSA verification function
won't accept any signature whose length doesn't match the RSA key
length. So on verify, I had to make a temporary buffer and pad the start
with zeroes to make it match the RSA key length.

Note that this wasn't a problem with the Public Key / UID/ Signature
packet signature, because we didn't precalculate the length of the
packet to write the header before streaming, and because we don't ever
verify that signature currently in the irongpg code.

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+# IronSSH - End-to-end secure file transfer
+
+This project is a fork of the *openssh/openssh-portable* project. While most of the project is unchanged, specific additions have been made to create new executables that provide end-to-end security when transferring files to remote servers.
```
Lets link to the openssh project home page or github page.  Hmmm.  Let's link to their home page: openssh.com.

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+# IronSSH - End-to-end secure file transfer
```
We should put our logo above this. And actually into the github project also.  This guy: https://ironcore-website-beta.herokuapp.com/products/ironsftp.  I can do that later after the new website is released.

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+
+While *sftp* and *scp* use *ssh* to keep files secure while they are being transferred over the network, once those files hit the remote server, they are no longer protected. The **ironsftp** executable provides additional security. When you put a file on the server using **ironsftp**, the file is encrypted before it is uploaded, and it stays that way on the server. When you get a file from the s ...
+
+The extension '.iron' is used to denote secured files. If you run **ironsftp** and put 'foo.c' on the server, it will be written remotely as 'foo.c.iron'. When you get 'foo.c', **ironsftp** will look for 'foo.c', but if it doesn't find that file, it will try to download 'foo.c.iron'. If that is successful, it will decrypt that file and write 'foo.c' on your local machine. The process operates the ...
```
Should probably use backticks instead of single quotes to surround file names and extensions.

zmre replied:
Also, maybe spell out that fetching `foo.c` getting `foo.c.iron` is just a convenience and the full filename can be specified for completeness or to resolve a conflict when both `foo.c` and `foo.c.iron` exist on the server.

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+
+## Key Management
+
+In order to use **ironsftp**, you must currently have an RSA key in the *~/.ssh/id_rsa* file on your local machine. When you start **ironsftp** the first time, it reads your public and private RSA keys (which may prompt you to enter the passphrase for the private key), then copies them into new key files under *~/.ssh/ironcore/*. The RSA key is used for signing encrypted files. **ironsftp** also  ...
```
Ditto here. Instead of italics, consider using backticks for the file paths. I think you should note at the end of this that your encryption keys are protected by your RSA authentication key. If your RSA key is locked, then so are your encryption keys. Unlocking your RSA key unlocks both.  But in better wording.

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+
+In order to use **ironsftp**, you must currently have an RSA key in the *~/.ssh/id_rsa* file on your local machine. When you start **ironsftp** the first time, it reads your public and private RSA keys (which may prompt you to enter the passphrase for the private key), then copies them into new key files under *~/.ssh/ironcore/*. The RSA key is used for signing encrypted files. **ironsftp** also  ...
+
+When you use **ironsftp** to connect to a server, your public key information is uploaded to `~/.ironpubkey`.
```
Maybe slightly more info, like "When you use **ironsftp** to connect to a server, your public key information is uploaded to `~/.ironpubkey` so other users can find it and share files with you."

zmre commented on README.md:
```diff
@@ -0,0 +1,63 @@
+
+## Secure Sharing
+
+You are also able to share these secure files with other users. When you connect to a server, by default, each file that you upload will be encrypted so that only you can read it. However, if other users on that server have connected to it using **ironsftp**, their public key information will be available in `~<login>/.ironpubkey`. You can use new **ironsftp** commands to add recipients, so that  ...
```
"any subsequent files you upload will be" -> "any subsequent files you upload in that session will be"

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+The options are as follows:
+.Bl -tag -width Ds
+.It Fl 1
+Specify the use of protocol version 1.
```
Gosh. This just reminds me that sshv1 is still around. Really tempting to strip that out of our stuff.

BobWall23 replied:
Why invite the merge hell?

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+Since it lacks user interaction it should be used in conjunction with
+non-interactive authentication.
+A
+.Ar batchfile
```
Have we tested our stuff in batch file mode?

BobWall23 replied:
Some. Just basic tests - connect, change dir, put, add recipient, put, change local dir, get.

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+.Xr ssh_config 5 .
+This is useful for specifying options
+for which there is no separate
+.Nm sftp
```
this shoud probably be `ironsftp`

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+.Nm sftp
+command-line flag.
+For example, to specify an alternate port use:
+.Ic sftp -oPort=24 .
```
Again, `ironsftp`

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+-specific command.)
+.It Ic bye
+Quit
+.Nm sftp .
```
`ironsftp`

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+extension.
+.It Ic exit
+Quit
+.Nm sftp .
```
`ironsftp`

zmre commented on ironsftp.1:
```diff
@@ -0,0 +1,675 @@
+.Xr sshd 8
+.Rs
+.%A T. Ylonen
+.%A S. Lehtinen
```
Add IronCore Labs, Inc.?  Or Bob Wall?

BobWall23 replied:
This is the reference to a publication written by those two people.

zmre commented:
You might want to force rebase this so the merge works nicely.  After you've made suggested changes.  With that done, I think this can be closed. I suspect we'll have more changes that we want to make as we go -- things to clarify and expand on -- but this is a great start.

BobWall23 committed:
Updates to doc files after code review.
